example/lfp/plot-lfp.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+
+import argparse
+import matplotlib.pyplot as plt
+import json
+import math
+import sys
+
+# Read JSON output of lfp example from stdin and plot.
+#
+# The JSON data for timeseries is structured as:
+#    <string>: { "unit": <string>, "time": [ <number>... ], "value": [ <number>... ] }
+# or:
+#    <string>: { "unit": <string>, "time": [ <number>... ], "values": [[ <number>... ] ...] }
+#
+# 2-d morphology data is represented as samples (x, z, r) where r is the radius, one array
+# per branch. Extra point data (probe location, electrode sites) as pairs (x, z):
+#   "morphology": { "unit": <string>, "samples":  [[[<number> <number> <number>] ...] ...]
+#                   "probe": [<number> <number>], "electrodes": [[<number> <number>] ...] }
+
+def subplot_timeseries(fig, index, jdict, key):
+    data = jdict[key]
+    sub = fig.add_subplot(index, ylabel=data['unit'], title=key, xlabel='time (ms)')
+    ts = data['time']
+    vss = data['values'] if 'values' in data else [data['value']]
+
+    for vs in vss: sub.plot(ts, vs)
+
+def subplot_morphology(fig, index, jdict, key, xlim, ylim):
+    data = jdict[key]
+    unit = data['unit']
+    sub = fig.add_subplot(index, xlabel='x ('+unit+')', ylabel='y ('+unit+')', title=key, xlim=xlim, ylim=ylim)
+
+    for samples in data['samples']:
+        spairs = zip(samples, samples[1:])
```

Comment:
I forgot that I'd moved this expression into the comprehension below. It can go too.